src/Services/ZeroRPCFactory.php

```diff
@@ -38,3 +40,44 @@ public function destroyClient($serviceName, $version, $timeout = null)
         unset(self::$_clients[$key]);
     }
 }
+
+/*
+ * When a client object is created, the PHP client will immediately connect
+ * to the endpoint. This wrapper avoids connecting until there is an actual
+ * function call.
+ */
+class _LazyClientWrapper
```

Comment:
Better to have another file for this class. Multiple class declarations in a single file, which violates the PSR-0/PSR-4 standards.